### DISCUSSION TITLE
Add local network multiplayer support

### DISCUSSION THREAD

**Discussion by lagios** [ppy/osu#13019] [Ideas]
I would like you to implement the multiplayer function in "LAN" mode without internet to play with friends that are connected to the same network, and thus be able to play in multiplayer in "LAN"

**Comment by peppy** [comment]
Can you describe a scenario where you don't have an internet connection, or what you hope to see achieved by this functionality?

I'd love to see this happen to some degree, although it'd probably still be hosted but with local routing features when available.

**Comment by lagios** [comment]
> Can you describe a scenario where you don't have an internet connection, or what you hope to see achieved by this functionality?
> 
> I'd love to see this happen to some degree, although it'd probably still be hosted but with local routing features when available.

What I'm trying to say is that people who do not have a connection to play ONLINE, can play in a local area network (LAN) with friends who are connected to the same network.

**Comment by etheoo98** [comment]
> Can you describe a scenario where you don't have an internet connection, or what you hope to see achieved by this functionality?
> 
I was at an internet cafe with a couple of friends about a month ago and I convinced them to play multiplayer but bancho wouldn't allow us to sign in. I forgot the error message but I think it had something to do with too many sign in requests being made from the same location.

**Comment by holly-hacker** [comment]
I imagine it would also be convenient in an environment where players frequently change, like during twitchcon where only a limited amount of computers are available.
